aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -372,14 +393,14 @@ impl AptosVM {
         Ok(())
     }
 
-    fn execute_modules<S: MoveResolverExt>(
+    fn execute_modules_ext<S: MoveResolverExt>(
```

Comment:
why do we need these _ext function when we do conversion in the execute_single_transaction?

Reply:
There are functions that still need to produce `TransactionOutput`, e.g. `simulate_signed_transaction` calls `execute_script_or_script_function`, `execute_modules` and `discard_error_vm_status`. At the same time, we have user transaction processing which calls the same functions but needs to produce `TransactionOutputExt`. My solution is to use `_ext` wrappers to ensure the right return type. Alternatively we can unwrap in-place (e.g. functions in `simulate_signed_transaction` would create the same `TransactionOutputExt` but then we only take the output and ignore anything else.

Reply:
these set of _ext is confusing to me, I think we should return TransactionOutputExt consistently and have caller to materialize aggregators if needed.